### PR TITLE
[dev v2.7] k8s november patch release

### PR DESCRIPTION
**Pull request by vardhaman22**
created pr against a custom branch names dev-v2.7-1122-patches (created from dev-v2.7) because OOB releases.
for issue: https://github.com/rancher/rancher/issues/39369

### CODE REVIEW

File: data/data.json [hunk 1]
```diff
@@ -10292,6 +10292,49 @@
    "aciGbpServerContainer": "noiro/gbp-server:5.2.3.2.1d150da",
    "aciOpflexServerContainer": "noiro/opflex-server:5.2.3.2.1d150da"
   },
+  "v1.22.16-rancher1-1": {
```
Comment (snasovich) [inline data/data.json:10295]: There probably won't be 1.22, see more info here: https://github.com/rancher/hyperkube/pull/291#issuecomment-1308896775

Comment (snasovich) [reply data/data.json:10295]: Disregard, 1.22.16 actually got released (https://github.com/kubernetes/kubernetes/releases/tag/v1.22.16)

**Comment by snasovich** [comment]
@vardhaman22 , there were couple additional merges to `dev-v2.7` since you've branched off from it. Are you planning to bring those changes in via this PR or a separate one?

**Comment by vardhaman22** [comment]
> @vardhaman22 , there were couple additional merges to `dev-v2.7` since you've branched off from it. Are you planning to bring those changes in via this PR or a separate one?

@snasovich the custom branch ```dev-v2.7-1122-patches``` is now in sync with dev-v2.7 
(sync done using PR: https://github.com/rancher/kontainer-driver-metadata/tree/dev-v2.7-1122-patches)

File: rke/k8s_rke_system_images.go [hunk 2]
```diff
@@ -8522,5 +8522,137 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
 			IngressBackend:            m("k8s.gcr.io/defaultbackend:1.4"),
 			MetricsServer:             m("gcr.io/google_containers/metrics-server-amd64:v0.3.1"),
 		},
+		// Enabled out of band post v2.7.0
+		"v1.22.16-rancher1-1": {
```
Comment (snasovich) [inline rke/k8s_rke_system_images.go]: We group by minor release, so `v1.22.16-rancher1-1` definition should follow the previous `v1.22.x-ranchery-z` definition. The same for newly introduced 1.23 and 1.24.

E.g. like here: https://github.com/rancher/kontainer-driver-metadata/pull/967

**Review by snasovich — CHANGES_REQUESTED** [review]
Please address posted comment. Otherwise LGTM.

File: rke/k8s_rke_system_images.go (hunk 2 above)
Comment (vardhaman22) [reply rke/k8s_rke_system_images.go]: done

**Review by snasovich — APPROVED** [review]
LGTM!

**Review by kinarashah — APPROVED** [review]
Versions match with `dev-v2.6-1122-patches`

**Pull request merged**